services/server/src/sourcify-chains.ts

```diff
@@ -212,18 +226,13 @@ for (const chain of allChains) {
   if (chainId in sourcifyChainsExtensions) {
     const sourcifyExtension = sourcifyChainsExtensions[chainId];
 
-    let rpc: (string | FetchRequestRPCWithHeaderEnvName)[] = [];
-    let rpcWithoutApiKeys: string[] = [];
-    let rpcWithApiKeyMasked: string[] = [];
-    let traceSupportedRPCs: TraceSupportedRPC[] | undefined = undefined;
+    let rpcs: SourcifyRpc[] = [];
     if (sourcifyExtension.rpc) {
-      ({ rpc, rpcWithoutApiKeys, rpcWithApiKeyMasked, traceSupportedRPCs } =
-        buildCustomRpcs(sourcifyExtension.rpc));
+      rpcs = buildCustomRpcs(sourcifyExtension.rpc);
```

Comment:
If I understood correctly how it works, previously we were setting in `buildCustomRpcs` the rpc object coming directly from `sourcifyExtension`. So in the code below (239-247), when we update the headerValue the object was updated both in `sourcifyExtension` and `rpcs` because it is a linked object. 

Now inside `buildCustomRpcs` we are creating a new object so the loop at lines 239-247 only updates the object in `sourcifyExtension` but not `rpcs`
```ts
[...]
 } else if (sourcifyRpc.type === "FetchRequest") {
[...]
      rpcs.push({
        rpc: fetchRequestRpc,
        urlWithoutApiKey: sourcifyRpc.url,
        maskedUrl: sourcifyRpc.url,
        traceSupport: sourcifyRpc.traceSupport,
      });
}
```

Reply:
I think that's true... very good catch!